### ISSUE TITLE
Allow for different weights for different targets

### ISSUE THREAD

**Issue by gaobhub** [sktime/pytorch-forecasting#1790]
Can you please make the TimeSeriesDataSet to accept a list of weights for each corresponding to a target?

**Comment by mkuiack** [comment]
I would second this and also ask to be able to send additional information to the loss function. 
Something like a flag whether a value is censured on the upside or downside. This would enable Tobit model style loss functions which for specific data points reduce penalty for over/under forecasting if for example sales were artificially constrained.

**Comment by fkiraly** [comment]
@gaobhub, @mkuiack, what would be much appreciated is a snippet of example code where you pass weights per target.

This would be passed as part of the `TimeSeriesDataSet`, right?

@phoeenniixx, is this already supported in your v2 redesign, or would this be easy to add? Not necessary in the first full prototype, but I was just thinking how we could easily support the "weights overall" and "weights per variable" cases both.

**Comment by phoeenniixx** [comment]
> @phoeenniixx, is this already supported in your v2 redesign, or would this be easy to add? Not necessary in the first full prototype, but I was just thinking how we could easily support the "weights overall" and "weights per variable" cases both.

Well for now I have not added these functionality, but it would be really good to have! I will read about it and would try to incorporate this in my design
(Not necessarily in next iteration, but maybe in the subsequent ones)

**Comment by gaobhub** [comment]
> [@gaobhub](https://github.com/gaobhub), [@mkuiack](https://github.com/mkuiack), what would be much appreciated is a snippet of example code where you pass weights per target.
> 
> This would be passed as part of the `TimeSeriesDataSet`, right?
> 
> [@phoeenniixx](https://github.com/phoeenniixx), is this already supported in your v2 redesign, or would this be easy to add? Not necessary in the first full prototype, but I was just thinking how we could easily support the "weights overall" and "weights per variable" cases both.

Yes, it is in `TimeSeriesDataSet`. Currently in `TimeSeriesDataSet`, the weight vector is shared by all targets if it is multi-target. I would suggest to make the weight the same dimension with targets for a multi-target case to allow for one weight corresponding to one target.
